### PR TITLE
Remove separate script for nightly devsite diff

### LINKED ISSUE #4066: Documentation code with dead link
**Description of the issue**

On this documentation page:
https://quantumai.google/reference/python/cirq/experiments/xeb_fitting/characterize_phased_fsim_parameters_with_xeb_by_pair
if you click "View source on GitHub" it does not go to a live link.

### LINKED ISSUE #4104: API reference for cirq.google is broken
**Description of the issue**

Objects under cirq.google are inaccessible currently since the v0.11 release.

**How to reproduce the issue**

E.g.: https://quantumai.google/reference/python/cirq/google/ConvertToXmonGates

**Cirq version**
v0.11

### PR DESCRIPTION
**Pull request by balopat**
After the v0.11 release there is no need for a separate script for the extra modules.
I also added the fix for the code_prefix_url which should fix the issue with the broken "View Source" functionality.
An unrelated nit that I changed is the default value for the site_url parameter - we had this URL structure when we first had the internal staging site hosted on tensorflow's site, now it's in line with what we have live. 

For the next module extractions this script will need to cater for both the stable (v0.11) and the nightly (from master) version, it will be more manageable that way.

Fixes #4066.
Fixes #4104.

### CODE REVIEW

File: dev_tools/docs/build_api_docs.py [hunk 1]
```diff
@@ -86,5 +92,25 @@ def main(unused_argv):
...
+        search_hints=FLAGS.search_hints,
+        site_path=FLAGS.site_path,
+        callbacks=[public_api.local_definitions_filter, filter_unwanted_inherited_methods],
+        private_map={
+            # Opt to not build docs for these paths for now since they error.
+            "cirq_google.engine.client.quantum.QuantumEngineServiceClient": ["enums"],
+            "cirq_google.engine.client.quantum_v1alpha1.QuantumEngineServiceClient": ["enums"],
+            "cirq_google.api": ["v1"],
+        },
+        extra_docs=_doc.RECORDED_CONST_DOCS,
```
Comment (95-martin-orion) [inline dev_tools/docs/build_api_docs.py:110]: Does `cirq_google` need this? It lives in `cirq-core`, and seems to be `cirq`-specific.

File: dev_tools/docs/build_api_docs.py [hunk 2]
```diff
@@ -29,19 +34,21 @@
...
     "code_url_prefix",
-    "https://github.com/quantumlib/cirq/tree/master/cirq",
+    "https://github.com/quantumlib/Cirq/blob/master",
     "The url prefix for links to code.",
 )
 
 flags.DEFINE_bool("search_hints", True, "Include metadata search hints in the generated files")
 
-flags.DEFINE_string("site_path", "quark/cirq/api_docs/python", "Path prefix in the _toc.yaml")
+flags.DEFINE_string("site_path", "reference/python", "Path prefix in the _toc.yaml")
```
Comment (95-martin-orion) [inline dev_tools/docs/build_api_docs.py:51]: What prompted this change? At first glance it doesn't seem related to either of the linked issues.

**Review by 95-martin-orion — APPROVED** [review]
A nit and a clarifying question, otherwise LGTM.

File: dev_tools/docs/build_api_docs.py (hunk 1 above)
Comment (balopat) [reply dev_tools/docs/build_api_docs.py:110]: Yes we need it here, this helps resolving docstrings for cirq-google public constants too e.g here: https://github.com/quantumlib/Cirq/blob/master/cirq-google/cirq_google/gate_sets.py#L70

File: dev_tools/docs/build_api_docs.py (hunk 2 above)
Comment (balopat) [reply dev_tools/docs/build_api_docs.py:51]: This is non-material, because the internal caller overrides this anyway with the correct string, but it was an outdated default value (we had this URL structure when we first had the internal staging site hosted on tensorflow's site) that I just updated. I'll include that in the PR description.
